lib/active_model/serializer/adapter/cached_serializer.rb

```diff
@@ -24,7 +24,7 @@ def cached?
         end
 
         def fragment_cached?
-          @klass._cache_only && !@klass._cache_except || !@klass._cache_only && @klass._cache_except
+          @klass._cache && (@klass._cache_only && !@klass._cache_except || !@klass._cache_only && @klass._cache_except)
```

Comment:
really, the block should check for `@klass._cache ` first, I think. `return yield unless @klass._cache` guard clause or something